lib/statsd.rb

```diff
@@ -125,9 +137,22 @@ def sampled(sample_rate)
 
   def send(stat, delta, type, sample_rate=1)
     sampled(sample_rate) do
-      prefix = "#{@namespace}." unless @namespace.nil?
-      stat = stat.to_s.gsub('::', '.').gsub(RESERVED_CHARS_REGEX, '_')
-      msg = "#{prefix}#{stat}:#{delta}|#{type}#{'|@' << sample_rate.to_s if sample_rate < 1}"
+      stat = stat.to_s.dup
+      stat.gsub!(COLON_COLON_REGEX, ".".freeze)
```

Comment:
I think I traced wrong earlier looking at the ruby source, but looking again it looks like a string will fall back to an index-based search using `rb_strseq_index`. But yeah, I guess we'll find out soon enough ;)